### PR TITLE
[RFC] General docs cleanup (1)

### PR DESCRIPTION
**Pull request by ghost**
refs #2231

There's still plenty more to do, but I'm trying to make these not too hard to review. I purposely didn't touch `vi_diff.txt` and any of the `pi_*.txt` stuff (netrw and whatnot), except to remove obsolete tags.

@justinmk I'm not sure if you still want the one-liner example included.

### CODE REVIEW

**Comment by justinmk** [comment]
The one-liner will be obsolete soon, but thanks :/

**Comment by Pyrohh** [comment]
It looks like I clobbered your comment, but I saw it so no worries.

File: runtime/doc/starting.txt [hunk 1]
```diff
@@ -529,8 +491,7 @@ X11 GUI support.  See |gui-resources|.
 ==============================================================================
 3. Initialization				*initialization* *startup*
 
-This section is about the non-GUI version of Vim.  See |gui-fork| for
-additional initialization when starting the GUI.
+This section is about the non-GUI version of Vim.
```
Comment (justinmk) [inline runtime/doc/starting.txt]: I think we can take this out.  Even if there are a million neovim UIs we should strive to keep the official UIs consistent.

File: runtime/doc/develop.txt [hunk 2]
```diff
@@ -135,10 +135,9 @@ VIM IS... FLEXIBLE					*design-flexible*
 
 Vim should make it easy for users to work in their preferred styles rather
 than coercing its users into particular patterns of work.  This can be for
-items with a large impact (e.g., the 'compatible' option) or for details.  The
-defaults are carefully chosen such that most users will enjoy using Vim as it
```
Comment (justinmk) [inline runtime/doc/develop.txt]: Hey look at that. More evidence that we're doing what Bram intended :)

**Comment by justinmk** [comment]
LGTM after comments

File: runtime/doc/starting.txt (hunk 1 above)
Comment (Pyrohh) [inline runtime/doc/starting.txt]: :+1:

**Comment by Pyrohh** [comment]
@justinmk Rebased and repushed.

I have a question regarding the reviewing of commits. In the future, would you like me to push a small review commit (the suggestions you pointed out were only on two lines), or just continue squashing things.

I'm thinking the first might be a better idea, reason being that you know you already reviewed everything else, so you could just look at my review commit and then squash it yourself. The downside I see is that you have to squash them yourself (but the benefit of not having to re-review is obvious).

**Comment by justinmk** [comment]
For documentation, usually don't worry about a review commit. For delicate code changes, I would probably make a review commit(s).

**Comment by justinmk** [comment]
woo! Thanks

**Comment by Pyrohh** [comment]
Thanks for reviewing, removing this cruft is always very satisfying :)

**Pull request merged**